src/stringfilter_type.h

```diff
@@ -59,6 +59,7 @@ struct StringFilter {
 	bool IsEmpty() const { return this->word_index.size() == 0; }
 
 	void ResetState();
+	void AddLine(std::string string);
```

Comment:
`const std::string &string` as `std::string string` makes a copy, which doesn't seem needed by the looks of it. And maybe rename the parameter to `str` to match the implementation.